### DISCUSSION TITLE
Connecting local PostgreSQL database - can't get it to work

### DISCUSSION THREAD

**Discussion by gooseman1721** [fief-dev/fief#89] [Q&A] [answered]
I'll preface this by admitting that I am a noob, so this might be the result of not having set up everything correctly. 

###  The issue 

I start Fief with:
 
`docker run --name fief-server -p 8000:8000 -d --env-file .env --add-host=db:<my local ip address> ghcr.io/fief-dev/fief:latest`

The .env file is:
```
SECRET=<my secret>
FIEF_CLIENT_ID=<client id>
FIEF_CLIENT_SECRET=<client secret>
ENCRYPTION_KEY=<encryption key>
PORT=8000
ROOT_DOMAIN=localhost:8000
FIEF_DOMAIN=localhost:8000
FIEF_MAIN_USER_EMAIL=<email>
FIEF_MAIN_USER_PASSWORD=<password>
CSRF_COOKIE_SECURE=False
LOGIN_SESSION_COOKIE_SECURE=False
SESSION_COOKIE_SECURE=False
FIEF_ADMIN_SESSION_COOKIE_SECURE=False
DATABASE_TYPE=POSTGRESQL
DATABASE_HOST=<my local ip address>
DATABASE_PORT=5432
DATABASE_USERNAME=fief
DATABASE_PASSWORD=fief
DATABASE_NAME=fief
```

When I open localhost:8000/admin/, the page is blank. When I run `docker logs <container name>` the last message is:

```
sqlalchemy.exc.InterfaceError: (sqlalchemy.dialects.postgresql.asyncpg.InterfaceError) <class 'asyncpg.exceptions._base.InterfaceError'>: cannot perform operation: another operation is in progress
[SQL: SELECT admin_session_tokens.id, admin_session_tokens.created_at, admin_session_tokens.updated_at, admin_session_tokens.token, admin_session_tokens.raw_tokens, admin_session_tokens.raw_userinfo
FROM admin_session_tokens
WHERE admin_session_tokens.token = %s]
[parameters: ('a1649a4a32dccd17a2601e3b71fa91ddfe69777cea8a9ed65813b29ffefcd96e',)]
(Background on this error at: https://sqlalche.me/e/14/rvf5) 
```

I don't know what to do from this point. Perhaps I set up my database incorrectly, it works with my backend with no (known) issues though. 

### Background

I installed Fief according to the quickstart guide with SQLite, that worked perfectly, admin dashboard worked and etc. For my app though I would like to use Postgres which I have set up locally on my machine, but I'm running into problems. At first, I couldn't even connect to it at all, but I have since fixed it by adding `--add-host=db:<my local ip address>` to `docker run` and appropriately modifying DATABASE_HOST in .env file.

**Comment by fief-bailiff** [comment]
Hail, @gooseman1721 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment]
EDIT: it seems there are weird issues with the latest version, I'll investigate.

**Comment by frankie567** [comment, marked as answer]
Escalating this into issue #90! I'm currently implementing a fix!

**Reply by gooseman1721** [reply]
Appreciate the quick response, looking forward to the fix!

**Reply by frankie567** [reply]
Should be fixed in [0.17.4](https://github.com/fief-dev/fief/releases/tag/v0.17.4) :tada: Make sure to do `docker pull ghcr.io/fief-dev/fief:latest` before to download the latest image.

Let me know if it's fixed on your side!

**Reply by gooseman1721** [reply]
With the newest image I could access /admin/ and successfully logged in, so that's fixed. Thanks!
